### ISSUE TITLE
JSON output status

### ISSUE THREAD

**Issue by EdwardNickson** [canonical/ubuntu-pro-client#2572]
Hi,
Just seeking some clarification around the `pro status` cli response. I'm looking at automating the `pro attach` process for a fleet of machines and I can see there is json and yaml output. Currently the response says "Content provided in json response is currently considered Experimental and may change", does this mean the yaml output is safer to parse, or is that just as experimental?

There is also the note at the bottom of [this documentation page](https://github.com/canonical/ubuntu-pro-client/blob/main/docs/explanations/status_columns.md) saying "`pro status --format=json` is the recommended machine-readable interface to this data"

**Comment by orndorffgrant** [comment]
Hello @EdwardNickson! Thanks for reaching out - good question.
The yaml output also contains the message: "Content provided in json response is currently considered Experimental and may change". The yaml output is exactly the same data as the json output, just formatted differently, so it is not better or worse than the json output.

Despite the message, we have several consumers of this output and we strive to maintain backwards compatibility with past versions of this output. We have added new fields and we recently added a new "status" value of "warning", but the structure is otherwise the same as it has been since this was last changed in mid-2021. We will continue to maintain backwards compatibility and consider it safe to rely on this output into the future.

The warning message is there because we have plans for a better interface to this data which will become the recommended interface once it exists. The new interface will have better versioning and an optional python API. Even after this exists, we will maintain `pro status --format json` because we already have many consumers of it and we won't break them. We may change the warning message to point to the newer interface, and we may print something similar to stderr, but we will avoid breaking the output. Any deprecation or removal of a feature if it happens will be aligned with an Ubuntu release: e.g. perhaps something could be removed in 24.04 onwards, but will still exist in all prior releases.
